### PR TITLE
Pydanticdocs2

### PR DESCRIPTION
**Pull request by loriab**
## Description
Switch from home-grown autodocs to pydantic autodocs. Fix some external entries in objects.inv. Retire custom sphinx theme. Needs #361 first, then a rebase.

## Status

- [x] Code base linted
- [x] Ready to go

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -97,11 +106,7 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-try:
-    import qcarchive_sphinx_theme
-    html_theme = 'qcarchive_sphinx_theme'
-except ModuleNotFoundError:
-    html_theme = 'sphinx_rtd_theme'
+html_theme = 'sphinx_rtd_theme'
```
Comment (dotsdl) [inline docs/source/conf.py:109]: Just curious, fine with the choice: is the [`qcarchive_sphinx_theme`](https://github.com/MolSSI/qcarchive-sphinx-theme) broken, unmaintained, or both?

**Review by dotsdl — APPROVED** [review]
Looks good to me @loriab! Just one question for clarification; approving changes.

File: docs/source/conf.py (hunk 1 above)
Comment (loriab) [reply docs/source/conf.py:109]: It's functional but receives little maintenance. It was producing some weird formatting (giant font when tiny font expected), and Ben's moving qcf to plain rtd anyways.

Comment (dotsdl) [reply docs/source/conf.py:109]: Ah, sounds good, no problem!

**Pull request merged**